dashboard/app/controllers/lti_v1_controller.rb

```diff
@@ -407,6 +408,90 @@ def new_integration
     render template: Policies::Lti.early_access? ? 'lti/v1/integrations/early_access' : 'lti/v1/integrations'
   end
 
+  # GET /lti/v1/dynamic_registration
+  # Initiate LTI Dynamic Registration flow
+  def new_registration
+    openid_configuration_url = params[:openid_configuration]
+    registration_token = params[:registration_token]
+    begin
+      openid_config = JSON.parse(HTTParty.get(openid_configuration_url).body).symbolize_keys
+    rescue => exception
+      Honeybadger.notify(exception, context: {message: 'Error getting LMS openid_configuration'})
+      return render status: :internal_server_error
+    end
+
+    registration_id = SecureRandom.uuid
+    puts registration_id
+    registration_data = {
+      registration_token: registration_token,
+      registration_endpoint: openid_config[:registration_endpoint],
+      issuer: openid_config[:issuer],
+      registration_id: registration_id,
+    }
+
+    write_cache(registration_id, registration_data, 1.hour)
```

Comment:
TODO: Add comment on why 1 hour, due to the TTL on the registration token we get from Canvas